Escape path for live reload. Fix #1247.

Aidiakapi opened the pull request:
Sanity check:

* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/getzola/zola/pulls) for the same update/change?

## Code changes

* [x] Are you doing the PR on the `next` branch?

This is a fix for #1247.

Keats commented on src/cmd/serve.rs:
```diff
@@ -154,13 +154,13 @@ fn rebuild_done_handling(broadcaster: &Option<Sender>, res: Result<()>, reload_p
                         "liveImg": true,
                         "protocol": ["http://livereload.com/protocols/official-7"]
                     }}"#,
-                        reload_path
```
I think we can write that one by hand, I don't really understand why this fixes the problem tbh

Aidiakapi replied:
The problem isn't the path, the path's fine, the problem is that `"D:\Some\Directory\path.html"` is not valid JSON. Because `\` like in Rust, is an escape sequence, so it fails to parse. It should become `"D:\\Some\\Directory\\path.html"`. I chose `serde_json` because it's already in the dependency hierarchy anyways (the templates use it).

Keats replied:
I see! Can you rebase? The tests should pass after that

Aidiakapi replied:
Ah yeah, didn't see the commit that snuck in there. It's done now.

Aidiakapi replied:
There's still something messed up in there. I'll look into it I guess. I'll just hard reset and patch.

Aidiakapi replied:
Alright, should be all good now.

Aidiakapi committed:
Escape path for live reload. Fix #1247.

zola serve's live reload feature used to fail on Windows due to the path
separator (`\`) to not being escaped.
